internal/netsync/manager.go

```diff
@@ -1335,6 +1416,18 @@ func (m *SyncManager) handleHeadersMsg(hmsg *headersMsg) {
 			log.Info("Syncing chain")
 			m.progressLogger.SetLastLogTime(time.Now())
 
+			// Request headers starting from the parent of the best known header
+			// for the local chain from any sync candidates that have not yet
+			// had their best known block discovered now that the initial
+			// headers sync process is complete.
+			for peer := range m.peers {
+				m.maybeResolveOrphanBlock(peer)
```

Comment:
> I'm completely new to this code but from a purely logical perspective it seems like it would be sensible to call `maybeResolveOrphanBlock` on all peers in `onInitialChainSyncDone`. Once the sync is known/suspected to be complete, it should be possible to resolve all orphan blocks announced by peers (if they are part of the best chain).
> 
> **EDIT:** Now that I am looking further, perhaps it doesn't matter because the `announcedOrphanBlock` field is only used **before** sync is completed, so updating it upon completion is unnecessary?

It's probably not entirely clear from this PR alone since nothing actually uses it yet, but ultimately it applies both before and after the initial chain sync is completed.  However, it isn't needed in `onInitialchainSyncDone` because of how the overall sync works.

For a quick recap, there are 2 phases.  The first phase is the initial headers sync and the second phase is the chain (block) sync.  `onInitialChainSyncDone` happens at the end of the second phase.

The first phase is where you're realistically going to see all orphans barring some kind of chain split, because you don't know all the headers yet, so any block announcement headers will necessarily not be able to connect and thus be orphans.

However, in the second phase, all headers are known (or at least believed to be known) and thus any announcements for new blocks that show up while the chain sync is taking place will realistically be known.  In the event they aren't known, attempting to resolve them once all the blocks are downloaded and linked will not gain you anything because there are no new headers involved.

Also, once phase one is done, new headers continue to be are dded and resolved in parallel with the chain (block) sync.

Notice that the code I'm commenting on resolves the orphans once the initial headers sync is done which is likely what you were mostly thinking about with your keen observation.